internal/cmdsearch/cmdsearch.go

```diff
@@ -86,21 +90,34 @@ func (r *SearchRunner) preRunE(c *cobra.Command, args []string) error {
 		c.Flag("by-value-regex").Changed {
 		return errors.Errorf(`only one of ["by-value", "by-value-regex"] can be provided`)
 	}
+	r.Writer = c.OutOrStdout()
 	return nil
 }
 
 func (r *SearchRunner) runE(c *cobra.Command, args []string) error {
 	e := runner.ExecuteCmdOnPkgs{
-		Writer:             c.OutOrStdout(),
+		Writer:             &bytes.Buffer{}, // dummy writer, runner need not print any info
```

Comment:
If we don't want to keep whatever is written to the `io.Writer`, we should use `ioutil.Discard`.